keras/utils/conv_utils.py

```diff
@@ -85,7 +85,20 @@ def normalize_tuple(value, n, name):
         error_msg += (f'including element {single_value} of '
                       f'type {type(single_value)}')
         raise ValueError(error_msg)
-    return value_tuple
+
+  if allow_zero:
+    qualified_values = list(filter(lambda x: x >= 0, value_tuple))
```

Comment:
err, shouldn't this just be [v for v in value_tuple if v >= 0]. Using list() filter() and lambda make this quite hard to read.

Reply:
done